lib/rules/no-extra-parens.js

```diff
@@ -232,6 +234,22 @@ module.exports = {
             return EXCEPT_COND_ASSIGN && node.test.type === "AssignmentExpression";
         }
 
+        /**
+         * Determines if a node test expression is allowed to have a parenthesised condition in ternary expression
+         * @param {ASTNode} node The node to be checked.
+         * @returns {boolean} True if the conditional in ternary expression can be parenthesised.
+         * @private
+         */
+        function isCondTernaryException(node) {
+            const filterType = new Set(["Identifier", "CallExpression"]);
```

Comment:
Since the `no-mixed-operators` rule requires parentheses only around binary expressions, to align these two rules the new option should only apply when the parenthesized node is a BinaryExpression node. For example, these parentheses should still not be allowed:

```js
/* eslint no-extra-parens: ["error", "all", { conditionalTernary: false }] */

(a.b) ? c : d // false negative

```

It would be better to check if the node is BinaryExpression than to check if it's not in a list of other expressions.

Reply:
`LogicalExpression` (`a || b`) should also be checked here. All types with multiple elements should be checked, but hopefully I'm not missing any others.